packages/dd-trace/src/config.js

```diff
@@ -13,85 +13,153 @@ class Config {
   constructor (options) {
     options = options || {}
 
-    const enabled = coalesce(options.enabled, platform.env('DD_TRACE_ENABLED'), true)
-    const profilingEnabled = coalesce(options.profiling, platform.env('DD_PROFILING_ENABLED'), false)
-    const debug = coalesce(options.debug, platform.env('DD_TRACE_DEBUG'), false)
-    const logInjection = coalesce(options.logInjection, platform.env('DD_LOGS_INJECTION'), false)
-    const url = coalesce(options.url, platform.env('DD_TRACE_AGENT_URL'), platform.env('DD_TRACE_URL'), null)
-    const site = coalesce(options.site, platform.env('DD_SITE'), 'datadoghq.com')
-    const hostname = coalesce(
+    this.tags = {}
+
+    // DD_TAGS
+    tagger.add(this.tags, platform.env('DD_TAGS'))
+    tagger.add(this.tags, platform.env('DD_TRACE_TAGS'))
+    tagger.add(this.tags, platform.env('DD_TRACE_GLOBAL_TAGS'))
+    tagger.add(this.tags, options.tags)
+
+    // DD_TRACE_ANALYTICS_ENABLED=false
+    const DD_TRACE_ANALYTICS_ENABLED = coalesce(
+      options.analytics,
+      platform.env('DD_TRACE_ANALYTICS_ENABLED'),
+      platform.env('DD_TRACE_ANALYTICS'),
+      false
+    )
+    // DD_PROFILING_ENABLED=false
+    const DD_PROFILING_ENABLED = coalesce(
+      options.profiling,
+      false
+    )
+    // DD_TRACE_<INTEGRATION>_ANALYTICS_ENABLED=false
+    // DD_TRACE_<INTEGRATION>_ANALYTICS_SAMPLE_RATE=1.0
+    // DD_TRACE_<INTEGRATIONS>_ENABLED=?
+    /*
+    Object.keys(platform.plugins)
+      .forEach((plugin) => {
+        const INTEGRATION = plugin.toUpperCase()
+        const analyticsEnabled = coalesce(
+          platform.env(`DD_TRACE_${INTEGRATION}_ANALYTICS_ENABLED`),
+          false
+        )
+        const analyticsSampleRate = coalesce(
+          platform.env(`DD_TRACE_${INTEGRATION}_ANALYTICS_SAMPLE_RATE`),
+          1.0
+        )
+        const enabled = coalesce(
+          platform.env(`DD_TRACE_${INTEGRATION}_ENABLED`),
+          true
+        )
+        use(plugin, {
```

Comment:
is this correct? i know `enabled` is an option but i'm not sure about the other two.

Reply:
The `analytics` option can have the following structures:

```js
analytics: true // enable with default sample rate (1.0)
analytics: 0.5 // enable with 0.5 sample rate
analytics: {
  `express.request`: true, // enable for `express.request` spans with default rate
  `express.middleware`: 0.5 // enable for `express.middleware` spans with 0.5 rate
}
```

Reply:
I wouldn't use `use` if we can avoid it as it could cause all sorts of issues and is planned to be deprecated.

Reply:
so it should be more like `analytics: coalesce(analyticsSampleRat, eanalyticsEnabled)`?

Reply:
I would keep the same approach as the global one (with and without the `_enabled` suffix).